packages/node_modules/@node-red/nodes/core/common/20-inject.html

```diff
@@ -168,21 +168,38 @@
         inputs:0,
         outputs:1,
         outputLabels: function(index) {
-            var lab = this.payloadType;
-            if (lab === "json") {
-                try {
-                    lab = typeof JSON.parse(this.payload);
-                    if (lab === "object") {
-                        if (Array.isArray(JSON.parse(this.payload))) { lab = "Array"; }
-                    }
-                } catch(e) {
-                    return this._("inject.label.invalid"); }
-            }
-            var name = "inject.label."+lab;
-            var label = this._(name);
-            if (name !== label) {
-                return label;
+            var lab = '';
+            
+            // if only payload and topic - display payload type
+            // if only one property - show it's type
+            // if more than one property (other than payload and topic) - show "x properties" where x is the number of properties.
+
+            // this.props will not be an array for legacy inject nodes until they are re-deployed
+            if (Array.isArray(this.props)) {
+                var propertyCount = this.props.length;
+                var payloadProperty = this.props.find(p => p.p === 'payload');
+                var topicProperty = this.props.find(p => p.p === 'topic' && p.vt === 'str');
+                
+                if (payloadProperty && topicProperty) {
+                    lab = payloadProperty.vt;
+                } else if (propertyCount > 1){
+                    lab = propertyCount + ' properties';
+                } else if(propertyCount === 1){
+                    lab = this.props[0].vt;
+                }
+                
+                // Friendly names
+                switch(lab){
```

Comment:
also note in existing code we use the i18n translation files to do the lookup for us... so we don't do it here
```
var name = "inject.label."+lab;
                var label = this._(name);
                if (name !== label) { lab = label; }
```

Reply:
Ahh i missed that. I need to fix so at least translations should be possible.